src/jit/codegencommon.cpp

```diff
@@ -10824,6 +10824,9 @@ CORINFO_CLASS_HANDLE Compiler::GetHfaClassHandle(GenTreePtr tree)
         case GT_ASG:
             assert(tree->gtOp.gtOp1->gtOper == GT_LCL_VAR || tree->gtOp.gtOp1->gtOper == GT_LCL_FLD);
             return GetHfaClassHandle(tree->gtOp.gtOp1);
+            
+        default:
+            break;
```

Comment:
For this kind of case, you should use unreached() - see the JIT coding conventions here: https://github.com/dotnet/coreclr/blob/master/Documentation/coding-guidelines/clr-jit-coding-conventions.md